tracer.go

```diff
@@ -264,7 +264,9 @@ func (t *tracer) Extract(
 // Close releases all resources used by the Tracer and flushes any remaining buffered spans.
 func (t *tracer) Close() error {
 	t.reporter.Close()
-	t.sampler.Close()
+	if sampler, ok := t.sampler.(io.Closer); ok {
+		sampler.Close()
```

Comment:
can you add a test that this actually gets called for remote sampler?  Because io.Closer returns error.